src/fd_monitor.rs

```diff
@@ -418,8 +417,10 @@ impl BackgroundFdMonitor {
                     // Note there is no risk of an ABA problem because ItemIDs are never recycled.
                     continue;
                 };
-                if fds.test(item.fd.as_raw_fd()) {
-                    item.service();
+                if let Some(fd) = &item.fd {
+                    if fds.test(fd.as_raw_fd()) {
+                        item.service();
```

Comment:
`is_some_and()` would fit better here IMO (reduce fd scope).
Though I guess that would break symmetry so it's not completely obvious

Reply:
addressed 5edeb09702c12db0d32c5809596257210bdca8b5